### ISSUE TITLE
i have some questions about backbone_loss,it seams backbones loss has always been 0 

### ISSUE THREAD

**Issue by paulrich1234** [deepmind/alphafold#307]
Dear guys 
     Thanks for the great works your guys have done for the world .
i have found some questions about the backbones loss ,
it seams the  following  codes :
 in https://github.com/deepmind/alphafold/blob/main/alphafold/model/folding.py#:~:text=0%2C%20None%2C%20None))-,fape_loss,-%3D%20fape_loss_fn(rigid_trajectory%2C%20gt_rigid

fape_loss = frame_aligned_point_error(rigid_trajectory, gt_rigid, backbone_mask,
                           rigid_trajectory.trans, gt_rigid.trans,
                           backbone_mask)

in https://github.com/deepmind/alphafold/blob/main/alphafold/model/all_atom.py#:~:text=Vecs%20(num_frames%2C%20num_positions)-,local_pred_pos,-%3D%20r3.rigids_mul_vecs(

replace pred_frames with rigid_trajectory and pred_positions with rigid_trajectory.trans

it becames the following 
  local_pred_pos = r3.rigids_mul_vecs(
      jax.tree_map(lambda r: r[:, None], r3.invert_rigids(rigid_trajectory)),
      jax.tree_map(lambda x: x[None, :], rigid_trajectory.trans))

i have check that no matter what rigid_trajectory is ,the  local_pred_pos will always be zeros via mathmatic formula  .

cause if R is rotation T is translation for rigid_trajectory. 
Then r3.invert_rigids(rigid_trajectory)'s R' =R.Tranpose and T' =-(R.Transpose)*T 
local_pred_pos =R'x+T' =R'*T+T' = R.Tranpose*T-(R.Transpose)*T   and it does not depend on which R and T always been zero 

that means the distantce of local_pred_pos, local_target_pos is always almost zeros too .

Am i Wrong ? 

Thank you for your help .

Richard Zhao

**Comment by AnyaP** [comment]
Hi Richard! Thanks for your question. 
Note that we are using None to add dimensions. So `local_target_pos` and `local_pred_pos` are matrices, while the listed derivation applies only to the diagonal, which will indeed equal to zeros.
Closing this issue for now, but feel free to reopen if this is seems unclear or you have further questions.
